spec/github_helper_spec.rb

```diff
@@ -171,4 +133,151 @@ def mock_comment(body: '<!-- REUSE_ID: test-id --> Test', user_id: 1234)
+      allow(client).to receive(:list_milestones).and_return(test_milestone)
+      expect(described_class.get_milestone(test_repo, 'release')).to eq({ title: 'release/10.1' })
+    end
+  end
+
+  describe 'create_milestone' do
+    let(:test_repo) { 'repo-test/project-test' }
+    let(:test_milestone_number) { '10.0' }
+    let(:test_milestone_duedate) { '2022-10-22T23:39:01Z' }
+    let(:client) do
+      instance_double(
+        Octokit::Client,
+        create_milestone: nil
+      )
+    end
+
+    before do
+      allow(described_class).to receive(:client).and_return(client)
+    end
+
+    it 'has the correct dates to code freeze without submission' do
+      comment = "Code freeze: October 22, 2022\nApp Store submission: November 15, 2022\nRelease: October 25, 2022\n"
+      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+
+      expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
+      create_milestone(need_submission: false, milestone_duration: 24, days_code_freeze: 3)
+    end
+
+    it 'has the correct dates to code freeze with submission' do
+      comment = "Code freeze: October 22, 2022\nApp Store submission: October 22, 2022\nRelease: October 25, 2022\n"
+      options = { due_on: '2022-10-22T12:00:00Z', description: comment }
+
+      expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
+      create_milestone(need_submission: true, milestone_duration: 19, days_code_freeze: 3)
+    end
+
+    def create_milestone(need_submission:, milestone_duration:, days_code_freeze:)
+      described_class.create_milestone(
+        test_repo,
+        test_milestone_number,
+        test_milestone_duedate.to_time.utc,
+        milestone_duration,
+        days_code_freeze,
+        need_submission
+      )
+    end
+  end
+
+  describe 'create_release' do
+    let(:test_repo) { 'repo-test/project-test' }
+    let(:test_tag) { '1.0' }
+    let(:test_target) { 'dummysha123456' }
+    let(:test_description) { 'Hey Im a Test Description' }
+    let(:client) do
+      instance_double(
+        Octokit::Client,
+        create_release: nil
+      )
+    end
+
+    before do
+      allow(described_class).to receive(:client).and_return(client)
+    end
+
+    it 'has the correct options' do
+      options = { body: test_description, draft: true, name: test_tag, prerelease: false, target_commitish: test_target }
+      expect(client).to receive(:create_release).with(test_repo, test_tag, options)
+      mockrelease
+    end
+
+    it 'upload the assets to the correct location' do
+      test_assets = 'test-file.xml'
+      test_url = '/test/url'
+
+      allow(client).to receive(:create_release).and_return({ url: test_url })
+      expect(client).to receive(:upload_asset).with(test_url, test_assets, { content_type: 'application/octet-stream' })
+      mockrelease([test_assets])
+    end
+
+    def mockrelease(assets = [])
```

Comment:
I like that you included a term in this method name to show that it's just a small helper method for the test to call the method. But I'm not sure `mock` is the right term, because it still calls the real method, and not a mock of it.
Besides, this naming doesn't respect snake_case naming conventions 😛 . Also, using named parameters would have been nice here too.

May I suggest something more like `call_create_release(assets: [])` or something similar?
Also, whatever naming you come up with for those small helper methods in each of your `describe` test sets here, it would be nice to make the naming consistent (e.g. for `create_milestone` helper method on line 207)

Reply:
Yeah, I decided to go by the name of the method, as the already existing [comment_on_pr](https://github.com/wordpress-mobile/release-toolkit/blob/6d474fe9faed59ac8259ff0866982c434c4de0d9/spec/github_helper_spec.rb#L161-L168) is doing right above, just to keep following the pattern already in place.